### PR TITLE
use with when calling TFRecordWriter

### PR DESCRIPTION
**Pull request by FirefoxMetzger**
TFRecordWriter supports __enter__ and __exit__ . 
Calling it through with is more pythonic and does cleanup in case something throws an uncaught exception.

### CODE REVIEW

**Review by qmick — APPROVED** [review]

**Comment by FirefoxMetzger** [comment]
@qmick Anything I can do to progress the this PR?
It seems that the TF Test Suite never finishes.

**Comment by qmick** [comment]
Don't worry about that. You just need to wait until a reviewer with write access  approve this PR, then he/she will start the test.

**Review by jhseu — APPROVED** [review]

**Pull request merged**
